File: src/main/java/com/epicdragonworld/gameserver/model/holders/ItemHolder.java
package com.epicdragonworld.gameserver.model.holders;

import com.epicdragonworld.gameserver.enums.ItemType;

/**
 * @author Pantelis Andrianakis
 */
public class ItemHolder
{
	private final int _itemId;
	private final int _slotId;
	private final ItemType _itemType;
	private final boolean _stackable;
	private final boolean _tradable;
	private final long _price;
	private final int _sta;
	private final int _str;
	private final int _dex;
	private final int _int;
	private final SkillHolder _skillHolder;
	
	public ItemHolder(int itemId, int slotId, ItemType itemType, boolean stackable, boolean tradable, long price, int stamina, int strength, int dexterity, int intelect, SkillHolder skillHolder)
	{
		_itemId = itemId;
		_slotId = slotId;
		_itemType = itemType;
		_stackable = stackable;
		_tradable = tradable;
		_price = price;
		_sta = stamina;
		_str = strength;
		_dex = dexterity;
		_int = intelect;
		_skillHolder = skillHolder;
	}
	
	public int getItemId()
	{
		return _itemId;
	}
	
	public int getSlotId()
	{
		return _slotId;
	}
	
	public ItemType getItemType()
	{
		return _itemType;
	}
	
	public boolean isStackable()
	{
		return _stackable;
	}
	
	public boolean isTradable()
	{
		return _tradable;
	}
	
	public long getPrice()
	{
		return _price;
	}
	
	public int getSTA()
	{
		return _sta;
	}
	
	public int getSTR()
	{
		return _str;
	}
	
	public int getDEX()
	{
		return _dex;
	}
	
	public int getINT()
	{
		return _int;
	}
	
	public SkillHolder getSkillHolder()
	{
		return _skillHolder;
	}
}

File: EpicDragonWorldServer/Holders/ItemHolder.cs
/**
 * Author: Pantelis Andrianakis
 * Date: May 5th 2019
 */
public class ItemHolder
{
    private readonly int itemId;
    private readonly ItemSlot itemSlot;
    private readonly ItemType itemType;
    private readonly bool stackable;
    private readonly bool tradable;
    private readonly int stamina;
    private readonly int strength;
    private readonly int dexterity;
    private readonly int intelect;
    private readonly SkillHolder skillHolder;

    public ItemHolder(int itemId, ItemSlot itemSlot, ItemType itemType, bool stackable, bool tradable, int stamina, int strength, int dexterity, int intelect, SkillHolder skillHolder)
    {
        this.itemId = itemId;
        this.itemSlot = itemSlot;
        this.itemType = itemType;
        this.stackable = stackable;
        this.tradable = tradable;
        this.stamina = stamina;
        this.strength = strength;
        this.dexterity = dexterity;
        this.intelect = intelect;
        this.skillHolder = skillHolder;
    }

    public int GetItemId()
    {
        return itemId;
    }

    public ItemSlot GetItemSlot()
    {
        return itemSlot;
    }

    public ItemType GetItemType()
    {
        return itemType;
    }

    public bool IsStackable()
    {
        return stackable;
    }

    public bool IsTradable()
    {
        return tradable;
    }

    public int GetSTA()
    {
        return stamina;
    }

    public int GetSTR()
    {
        return strength;
    }

    public int GetDEX()
    {
        return dexterity;
    }

    public int GetINT()
    {
        return intelect;
    }

    public SkillHolder GetSkillHolder()
    {
        return skillHolder;
    }
}
